perl-5.38 compilation error (regcomp.o and lib/auto/re/re.a(re_comp.o) conflict)

###Description
at Makefile' compilation point $(PERL_EXE):
regcomp.o and  lib/auto/re/re.a(re_comp.o) have the same functions, that cause exception while compiling perl-5.38

### Steps to Reproduce
./Configure -de -Dprefix=/tools/perl-2.38 -Dusethreads

**compilation need libpthread**
sed -i 's|\\(\\$[(]miniperl_objs[)] \\$[(]libs[)]\\)$|\1 -lpthread|' Makefile
i=$(sed -n '/\\$[(]SHRPENV[)] \\$[(]CC[)]/=')
sed -i "$i"'s|\\(\\$[(]libs[)]\\)$|\1 -lpthread|' Makefile

**Encode modules like Encode-Byte cannot be found in ext/$spec, because they are in cpan/$spec**
i=$(sed -n "/Can't find extension/=" make_ext.pl); ((i--))
str='    } elsif (-d "cpan/$spec") {
	    # Encode-Byte => Encode/Byte
	    $ext_pathname = "cpan/$spec";'
bfl::insert_string_to_file "$str" "$i" make_ext.pl

**make**

**Expected behavior**
cc -o perl -fstack-protector-strong -L/usr/local/lib  perlmain.o  lib/auto/attributes/attributes.a lib/auto/B/B.a lib/auto/Compress/Raw/Bzip2/Bzip2.a lib/auto/Compress/Raw/Zlib/Zlib.a lib/auto/Cwd/Cwd.a lib/auto/Data/Dumper/Dumper.a lib/auto/Devel/Peek/Peek.a lib/auto/Devel/PPPort/PPPort.a lib/auto/Digest/MD5/MD5.a lib/auto/Digest/SHA/SHA.a lib/auto/Encode/Encode.a lib/auto/Fcntl/Fcntl.a lib/auto/File/DosGlob/DosGlob.a lib/auto/File/Glob/Glob.a lib/auto/Filter/Util/Call/Call.a lib/auto/Hash/Util/Util.a lib/auto/Hash/Util/FieldHash/FieldHash.a lib/auto/I18N/Langinfo/Langinfo.a lib/auto/IO/IO.a lib/auto/IPC/SysV/SysV.a lib/auto/List/Util/Util.a lib/auto/Math/BigInt/FastCalc/FastCalc.a lib/auto/MIME/Base64/Base64.a lib/auto/mro/mro.a lib/auto/Opcode/Opcode.a lib/auto/PerlIO/encoding/encoding.a lib/auto/PerlIO/mmap/mmap.a lib/auto/PerlIO/scalar/scalar.a lib/auto/PerlIO/via/via.a lib/auto/POSIX/POSIX.a lib/auto/re/re.a lib/auto/SDBM_File/SDBM_File.a lib/auto/Socket/Socket.a lib/auto/Storable/Storable.a lib/auto/Sys/Hostname/Hostname.a lib/auto/Sys/Syslog/Syslog.a lib/auto/threads/threads.a lib/auto/threads/shared/shared.a lib/auto/Time/HiRes/HiRes.a lib/auto/Time/Piece/Piece.a lib/auto/Unicode/Collate/Collate.a lib/auto/Unicode/Normalize/Normalize.a lib/auto/Encode/Byte/Byte.a lib/auto/Encode/CN/CN.a lib/auto/Encode/EBCDIC/EBCDIC.a lib/auto/Encode/JP/JP.a lib/auto/Encode/KR/KR.a lib/auto/Encode/Symbol/Symbol.a lib/auto/Encode/TW/TW.a lib/auto/Encode/Unicode/Unicode.a libperl.a `cat ext.libs` -lm -lc  -lpthread
/usr/bin/ld: libperl.a(regcomp.o): in function «Perl_reg_add_data»:
regcomp.c:(.text+0x6cb0): повторное определение «Perl_reg_add_data»; lib/auto/re/re.a(re_comp.o):re_comp.c:(.text+0xa000): здесь первое определение
/usr/bin/ld: libperl.a(regcomp.o): in function «Perl_populate_anyof_bitmap_from_invlist»:
regcomp.c:(.text+0x7760): повторное определение «Perl_populate_anyof_bitmap_from_invlist»; lib/auto/re/re.a(re_comp.o):re_comp.c:(.text+0xa5e0): здесь первое определение
/usr/bin/ld: libperl.a(regcomp.o): in function «Perl_add_above_Latin1_folds»:
regcomp.c:(.text+0x7790): повторное определение «Perl_add_above_Latin1_folds»; lib/auto/re/re.a(re_comp.o):re_comp.c:(.text+0xa950): здесь первое определение
/usr/bin/ld: libperl.a(regcomp.o): in function «Perl_set_ANYOF_arg»:
regcomp.c:(.text+0x7c50): повторное определение «Perl_set_ANYOF_arg»; lib/auto/re/re.a(re_comp.o):re_comp.c:(.text+0xae60): здесь первое определение
/usr/bin/ld: libperl.a(regcomp.o): in function «Perl_get_ANYOFM_contents»:
regcomp.c:(.text+0x8190): повторное определение «Perl_get_ANYOFM_contents»; lib/auto/re/re.a(re_comp.o):re_comp.c:(.text+0xb770): здесь первое определение
/usr/bin/ld: libperl.a(regcomp.o): in function «Perl_get_ANYOFHbbm_contents»:
regcomp.c:(.text+0x8250): повторное определение «Perl_get_ANYOFHbbm_contents»; lib/auto/re/re.a(re_comp.o):re_comp.c:(.text+0xb830): здесь первое определение
/usr/bin/ld: libperl.a(regcomp_invlist.o): in function «Perl_populate_bitmap_from_invlist»:
regcomp_invlist.c:(.text+0x360): повторное определение «Perl_populate_bitmap_from_invlist»; lib/auto/re/re.a(re_comp_invlist.o):re_comp_invlist.c:(.text+0x0): здесь первое определение
/usr/bin/ld: libperl.a(regcomp_invlist.o): in function «Perl_populate_invlist_from_bitmap»:
regcomp_invlist.c:(.text+0x1520): повторное определение «Perl_populate_invlist_from_bitmap»; lib/auto/re/re.a(re_comp_invlist.o):re_comp_invlist.c:(.text+0x260): здесь первое определение
collect2: error: ld returned 1 exit status
make: *** [Makefile:393: perl] Ошибка 

**Perl configuration**
Summary of my perl5 (revision 5 version 30 subversion 0) configuration:
   
  Platform:
    osname=linux
    osvers=4.19.0
    archname=x86_64-linux-gnu-thread-multi
    uname='linux localhost 4.19.0 #1 smp debian 4.19.0 x86_64 gnulinux '
    config_args='-Dmksymlinks -Dusethreads -Duselargefiles -Dcc=x86_64-linux-gnu-gcc -Dcpp=x86_64-linux-gnu-cpp -Dld=x86_64-linux-gnu-gcc -Dccflags=-DDEBIAN -Wdate-time -D_FORTIFY_SOURCE=2 -g -O2 -fdebug-prefix-map=/build/perl-Wfb2Cd/perl-5.30.0=. -fstack-protector-strong -Wformat -Werror=format-security -Dldflags= -Wl,-Bsymbolic-functions -Wl,-z,relro -Dlddlflags=-shared -Wl,-Bsymbolic-functions -Wl,-z,relro -Dcccdlflags=-fPIC -Darchname=x86_64-linux-gnu -Dprefix=/usr -Dprivlib=/usr/share/perl/5.30 -Darchlib=/usr/lib/x86_64-linux-gnu/perl/5.30 -Dvendorprefix=/usr -Dvendorlib=/usr/share/perl5 -Dvendorarch=/usr/lib/x86_64-linux-gnu/perl5/5.30 -Dsiteprefix=/usr/local -Dsitelib=/usr/local/share/perl/5.30.0 -Dsitearch=/usr/local/lib/x86_64-linux-gnu/perl/5.30.0 -Dman1dir=/usr/share/man/man1 -Dman3dir=/usr/share/man/man3 -Dsiteman1dir=/usr/local/man/man1 -Dsiteman3dir=/usr/local/man/man3 -Duse64bitint -Dman1ext=1 -Dman3ext=3perl -Dpager=/usr/bin/sensible-pager -Uafs -Ud_csh -Ud_ualarm -Uusesfio -Uusenm -Ui_libutil -Ui_xlocale -Uversiononly -DDEBUGGING=-g -Doptimize=-O2 -dEs -Duseshrplib -Dlibperl=libperl.so.5.30.0'
    hint=recommended
    useposix=true
    d_sigaction=define
    useithreads=define
    usemultiplicity=define
    use64bitint=define
    use64bitall=define
    uselongdouble=undef
    usemymalloc=n
    default_inc_excludes_dot=define
    bincompat5005=undef
  Compiler:
    cc='x86_64-linux-gnu-gcc'
    ccflags ='-D_REENTRANT -D_GNU_SOURCE -DDEBIAN -fwrapv -fno-strict-aliasing -pipe -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64'
    optimize='-O2 -g'
    cppflags='-D_REENTRANT -D_GNU_SOURCE -DDEBIAN -fwrapv -fno-strict-aliasing -pipe -I/usr/local/include'
    ccversion=''
    gccversion='9.3.0'
    gccosandvers=''
    intsize=4
    longsize=8
    ptrsize=8
    doublesize=8
    byteorder=12345678
    doublekind=3
    d_longlong=define
    longlongsize=8
    d_longdbl=define
    longdblsize=16
    longdblkind=3
    ivtype='long'
    ivsize=8
    nvtype='double'
    nvsize=8
    Off_t='off_t'
    lseeksize=8
    alignbytes=8
    prototype=define
  Linker and Libraries:
    ld='x86_64-linux-gnu-gcc'
    ldflags =' -fstack-protector-strong -L/usr/local/lib'
    libpth=/usr/local/lib /usr/include/x86_64-linux-gnu /usr/lib /lib/x86_64-linux-gnu /lib/../lib /usr/lib/x86_64-linux-gnu /usr/lib/../lib /lib
    libs=-lgdbm -lgdbm_compat -ldb -ldl -lm -lpthread -lc -lcrypt
    perllibs=-ldl -lm -lpthread -lc -lcrypt
    libc=libc-2.31.so
    so=so
    useshrplib=true
    libperl=libperl.so.5.30
    gnulibc_version='2.31'
  Dynamic Linking:
    dlsrc=dl_dlopen.xs
    dlext=so
    d_dlsymun=undef
    ccdlflags='-Wl,-E'
    cccdlflags='-fPIC'
    lddlflags='-shared -L/usr/local/lib -fstack-protector-strong'


Characteristics of this binary (from libperl): 
  Compile-time options:
    HAS_TIMES
    MULTIPLICITY
    PERLIO_LAYERS
    PERL_COPY_ON_WRITE
    PERL_DONT_CREATE_GVSV
    PERL_IMPLICIT_CONTEXT
    PERL_MALLOC_WRAP
    PERL_OP_PARENT
    PERL_PRESERVE_IVUV
    USE_64_BIT_ALL
    USE_64_BIT_INT
    USE_ITHREADS
    USE_LARGE_FILES
    USE_LOCALE
    USE_LOCALE_COLLATE
    USE_LOCALE_CTYPE
    USE_LOCALE_NUMERIC
    USE_LOCALE_TIME
    USE_PERLIO
    USE_PERL_ATOF
    USE_REENTRANT_API
    USE_THREAD_SAFE_LOCALE
  Locally applied patches:
    DEBPKG:debian/cpan_definstalldirs - Provide a sensible INSTALLDIRS default for modules installed from CPAN.
    DEBPKG:debian/db_file_ver - https://bugs.debian.org/340047 Remove overly restrictive DB_File version check.
    DEBPKG:debian/doc_info - Replace generic man(1) instructions with Debian-specific information.
    DEBPKG:debian/enc2xs_inc - https://bugs.debian.org/290336 Tweak enc2xs to follow symlinks and ignore missing @INC directories.
    DEBPKG:debian/errno_ver - https://bugs.debian.org/343351 Remove Errno version check due to upgrade problems with long-running processes.
    DEBPKG:debian/libperl_embed_doc - https://bugs.debian.org/186778 Note that libperl-dev package is required for embedded linking
    DEBPKG:fixes/respect_umask - Respect umask during installation
    DEBPKG:debian/writable_site_dirs - Set umask approproately for site install directories
    DEBPKG:debian/extutils_set_libperl_path - EU:MM: set location of libperl.a under /usr/lib
    DEBPKG:debian/no_packlist_perllocal - Don't install .packlist or perllocal.pod for perl or vendor
    DEBPKG:debian/fakeroot - Postpone LD_LIBRARY_PATH evaluation to the binary targets.
    DEBPKG:debian/instmodsh_doc - Debian policy doesn't install .packlist files for core or vendor.
    DEBPKG:debian/ld_run_path - Remove standard libs from LD_RUN_PATH as per Debian policy.
    DEBPKG:debian/libnet_config_path - Set location of libnet.cfg to /etc/perl/Net as /usr may not be writable.
    DEBPKG:debian/perlivp - https://bugs.debian.org/510895 Make perlivp skip include directories in /usr/local
    DEBPKG:debian/squelch-locale-warnings - https://bugs.debian.org/508764 Squelch locale warnings in Debian package maintainer scripts
    DEBPKG:debian/patchlevel - https://bugs.debian.org/567489 List packaged patches for 5.30.0-9ubuntu0.2 in patchlevel.h
    DEBPKG:fixes/document_makemaker_ccflags - https://bugs.debian.org/628522 [rt.cpan.org #68613] Document that CCFLAGS should include $Config{ccflags}
    DEBPKG:debian/find_html2text - https://bugs.debian.org/640479 Configure CPAN::Distribution with correct name of html2text
    DEBPKG:debian/perl5db-x-terminal-emulator.patch - https://bugs.debian.org/668490 Invoke x-terminal-emulator rather than xterm in perl5db.pl
    DEBPKG:debian/cpan-missing-site-dirs - https://bugs.debian.org/688842 Fix CPAN::FirstTime defaults with nonexisting site dirs if a parent is writable
    DEBPKG:fixes/memoize_storable_nstore - [rt.cpan.org #77790] https://bugs.debian.org/587650 Memoize::Storable: respect 'nstore' option not respected
    DEBPKG:debian/makemaker-pasthru - https://bugs.debian.org/758471 Pass LD settings through to subdirectories
    DEBPKG:debian/makemaker-manext - https://bugs.debian.org/247370 Make EU::MakeMaker honour MANnEXT settings in generated manpage headers
    DEBPKG:debian/kfreebsd-softupdates - https://bugs.debian.org/796798 Work around Debian Bug#796798
    DEBPKG:fixes/autodie-scope - https://bugs.debian.org/798096 Fix a scoping issue with "no autodie" and the "system" sub
    DEBPKG:fixes/memoize-pod - [rt.cpan.org #89441] Fix POD errors in Memoize
    DEBPKG:debian/hurd-softupdates - https://bugs.debian.org/822735 Fix t/op/stat.t failures on hurd
    DEBPKG:fixes/math_complex_doc_great_circle - https://bugs.debian.org/697567 [rt.cpan.org #114104] Math::Trig: clarify definition of great_circle_midpoint
    DEBPKG:fixes/math_complex_doc_see_also - https://bugs.debian.org/697568 [rt.cpan.org #114105] Math::Trig: add missing SEE ALSO
    DEBPKG:fixes/math_complex_doc_angle_units - https://bugs.debian.org/731505 [rt.cpan.org #114106] Math::Trig: document angle units
    DEBPKG:fixes/cpan_web_link - https://bugs.debian.org/367291 CPAN: Add link to main CPAN web site
    DEBPKG:debian/hppa_op_optimize_workaround - https://bugs.debian.org/838613 Temporarily lower the optimization of op.c on hppa due to gcc-6 problems
    DEBPKG:debian/installman-utf8 - https://bugs.debian.org/840211 Generate man pages with UTF-8 characters
    DEBPKG:fixes/getopt-long-4 - https://bugs.debian.org/864544 [rt.cpan.org #122068] Fix issue #122068.
    DEBPKG:debian/hppa_opmini_optimize_workaround - https://bugs.debian.org/869122 Lower the optimization level of opmini.c on hppa
    DEBPKG:debian/sh4_op_optimize_workaround - https://bugs.debian.org/869373 Also lower the optimization level of op.c and opmini.c on sh4
    DEBPKG:debian/perldoc-pager - https://bugs.debian.org/870340 [rt.cpan.org #120229] Fix perldoc terminal escapes when sensible-pager is less
    DEBPKG:debian/prune_libs - https://bugs.debian.org/128355 Prune the list of libraries wanted to what we actually need.
    DEBPKG:debian/mod_paths - Tweak @INC ordering for Debian
    DEBPKG:debian/configure-regen - https://bugs.debian.org/762638 Regenerate Configure et al. after probe unit changes
    DEBPKG:debian/deprecate-with-apt - https://bugs.debian.org/747628 Point users to Debian packages of deprecated core modules
    DEBPKG:debian/disable-stack-check - https://bugs.debian.org/902779 [perl #133327] Disable debugperl stack extension checks for binary compatibility with perl
    DEBPKG:fixes/eumm-usrmerge - https://bugs.debian.org/913637 Avoid mangling /bin non-perl shebangs on merged-/usr systems
    DEBPKG:debian/perlbug-editor - https://bugs.debian.org/922609 Use "editor" as the default perlbug editor, as per Debian policy
    DEBPKG:fixes/gid-parsing - [79e302e] https://bugs.debian.org/941985 [perl #134169] (perl #134169) mg.c reset endptr after use
    DEBPKG:fixes/CVE-2020-10543.patch - [PATCH v530 1/4] regcomp.c: Prevent integer overflow from nested regex quantifiers.
    DEBPKG:fixes/CVE-2020-10878-1.patch - [PATCH v530 2/4] study_chunk: extract rck_elide_nothing
    DEBPKG:fixes/CVE-2020-10878-2.patch - [PATCH v530 3/4] regcomp: use long jumps if there is any possibility of overflow
    DEBPKG:fixes/CVE-2020-12723.patch - [PATCH v530 4/4] study_chunk: avoid mutating regexp program within GOSUB
  Built under linux
  Compiled at Oct 19 2020 10:56:54
  @INC:
    /etc/perl
    /usr/local/lib/x86_64-linux-gnu/perl/5.30.0
    /usr/local/share/perl/5.30.0
    /usr/lib/x86_64-linux-gnu/perl5/5.30
    /usr/share/perl5
    /usr/lib/x86_64-linux-gnu/perl/5.30
    /usr/share/perl/5.30
    /usr/local/lib/site_perl
    /usr/lib/x86_64-linux-gnu/perl-base

I can duplicate the multiple definition errors with:
```
./Configure -des -Dusedevel -Dstatic_ext='re'
```
with blead:
```
cc -o perl -fstack-protector-strong -L/usr/local/lib -Wl,-E perlmain.o  lib/auto/re/re.a libperl.a `cat ext.libs` -lpthread -ldl -lm -lcrypt -lutil -lc 
/usr/bin/ld: libperl.a(regcomp.o): in function `Perl_reg_add_data':
regcomp.c:(.text+0x7d60): multiple definition of `Perl_reg_add_data'; lib/auto/re/re.a(re_comp.o):re_comp.c:(.text+0xa9f0): first defined here
/usr/bin/ld: libperl.a(regcomp.o): in function `Perl_populate_anyof_bitmap_from_invlist':
regcomp.c:(.text+0x8670): multiple definition of `Perl_populate_anyof_bitmap_from_invlist'; lib/auto/re/re.a(re_comp.o):re_comp.c:(.text+0xafa0): first defined here
/usr/bin/ld: libperl.a(regcomp.o): in function `Perl_add_above_Latin1_folds':
regcomp.c:(.text+0x8830): multiple definition of `Perl_add_above_Latin1_folds'; lib/auto/re/re.a(re_comp.o):re_comp.c:(.text+0xb280): first defined here
/usr/bin/ld: libperl.a(regcomp.o): in function `Perl_set_ANYOF_arg':
regcomp.c:(.text+0x8c90): multiple definition of `Perl_set_ANYOF_arg'; lib/auto/re/re.a(re_comp.o):re_comp.c:(.text+0xb750): first defined here
/usr/bin/ld: libperl.a(regcomp.o): in function `Perl_get_ANYOFM_contents':
regcomp.c:(.text+0xb020): multiple definition of `Perl_get_ANYOFM_contents'; lib/auto/re/re.a(re_comp.o):re_comp.c:(.text+0xdde0): first defined here
/usr/bin/ld: libperl.a(regcomp.o): in function `Perl_get_ANYOFHbbm_contents':
regcomp.c:(.text+0xb0e0): multiple definition of `Perl_get_ANYOFHbbm_contents'; lib/auto/re/re.a(re_comp.o):re_comp.c:(.text+0xdea0): first defined here
/usr/bin/ld: libperl.a(regcomp_invlist.o): in function `Perl_populate_bitmap_from_invlist':
regcomp_invlist.c:(.text+0x2e0): multiple definition of `Perl_populate_bitmap_from_invlist'; lib/auto/re/re.a(re_comp_invlist.o):re_comp_invlist.c:(.text+0x60): first defined here
/usr/bin/ld: libperl.a(regcomp_invlist.o): in function `Perl_populate_invlist_from_bitmap':
regcomp_invlist.c:(.text+0x1350): multiple definition of `Perl_populate_invlist_from_bitmap'; lib/auto/re/re.a(re_comp_invlist.o):re_comp_invlist.c:(.text+0x260): first defined here
collect2: error: ld returned 1 exit status
```

i found the same problem building in openwrt perl 5.38.0

```
./Configure -der -Uusedl -Duserelocatableinc -Dusethreads

./Configure -S
```

```
Can't find extension Encode/Symbol in any of cpan dist ext at make_ext.pl line 264.
./miniperl -Ilib -I. make_ext.pl lib/auto/Encode/TW/TW.a  MAKE="make" LIBPERL_A=libperl.a LINKTYPE=static CCCDLFLAGS=
Can't find extension Encode/TW in any of cpan dist ext at make_ext.pl line 264.
./miniperl -Ilib -I. make_ext.pl lib/auto/Encode/Unicode/Unicode.a  MAKE="make" LIBPERL_A=libperl.a LINKTYPE=static CCCDLFLAGS=
Can't find extension Encode/Unicode in any of cpan dist ext at make_ext.pl line 264.
cc -o perl -fstack-protector-strong -L/usr/local/lib  perlmain.o  lib/auto/B/B.a lib/auto/Compress/Raw/Bzip2/Bzip2.a lib/auto/Compress/Raw/Zlib/Zlib.a lib/auto/Cwd/Cwd.a lib/auto/Data/Dumper/Dumper.a lib/auto/Devel/PPPort/PPPort.a lib/auto/Devel/Peek/Peek.a lib/auto/Digest/MD5/MD5.a lib/auto/Digest/SHA/SHA.a lib/auto/Encode/Encode.a lib/auto/Fcntl/Fcntl.a lib/auto/File/DosGlob/DosGlob.a lib/auto/File/Glob/Glob.a lib/auto/Filter/Util/Call/Call.a lib/auto/Hash/Util/Util.a lib/auto/Hash/Util/FieldHash/FieldHash.a lib/auto/I18N/Langinfo/Langinfo.a lib/auto/IO/IO.a lib/auto/IPC/SysV/SysV.a lib/auto/List/Util/Util.a lib/auto/MIME/Base64/Base64.a lib/auto/Math/BigInt/FastCalc/FastCalc.a lib/auto/Opcode/Opcode.a lib/auto/POSIX/POSIX.a lib/auto/PerlIO/encoding/encoding.a lib/auto/PerlIO/mmap/mmap.a lib/auto/PerlIO/scalar/scalar.a lib/auto/PerlIO/via/via.a lib/auto/SDBM_File/SDBM_File.a lib/auto/Socket/Socket.a lib/auto/Storable/Storable.a lib/auto/Sys/Hostname/Hostname.a lib/auto/Sys/Syslog/Syslog.a lib/auto/Time/HiRes/HiRes.a lib/auto/Time/Piece/Piece.a lib/auto/Unicode/Collate/Collate.a lib/auto/Unicode/Normalize/Normalize.a lib/auto/attributes/attributes.a lib/auto/mro/mro.a
lib/auto/re/re.a lib/auto/threads/threads.a lib/auto/threads/shared/shared.a lib/auto/Encode/Byte/Byte.a lib/auto/Encode/CN/CN.a lib/auto/Encode/EBCDIC/EBCDIC.a lib/auto/Encode/JP/JP.a lib/auto/Encode/KR/KR.a lib/auto/Encode/Symbol/Symbol.a lib/auto/Encode/TW/TW.a lib/auto/Encode/Unicode/Unicode.a libperl.a `cat ext.libs` -lpthread -ldl -lm -lcrypt -lutil -lc
/usr/bin/ld: libperl.a(regcomp.o): in function `Perl_reg_add_data':
regcomp.c:(.text+0x6c40): multiple definition of `Perl_reg_add_data'; lib/auto/re/re.a(re_comp.o):re_comp.c:(.text+0x9520): first defined here
/usr/bin/ld: libperl.a(regcomp.o): in function `Perl_populate_anyof_bitmap_from_invlist':
regcomp.c:(.text+0x7730): multiple definition of `Perl_populate_anyof_bitmap_from_invlist'; lib/auto/re/re.a(re_comp.o):re_comp.c:(.text+0x9af0): first defined here
/usr/bin/ld: libperl.a(regcomp.o): in function `Perl_add_above_Latin1_folds':
regcomp.c:(.text+0x7750): multiple definition of `Perl_add_above_Latin1_folds'; lib/auto/re/re.a(re_comp.o):re_comp.c:(.text+0x9db0): first defined here
/usr/bin/ld: libperl.a(regcomp.o): in function `Perl_set_ANYOF_arg':
regcomp.c:(.text+0x7bf0): multiple definition of `Perl_set_ANYOF_arg'; lib/auto/re/re.a(re_comp.o):re_comp.c:(.text+0xa2c0): first defined here
/usr/bin/ld: libperl.a(regcomp.o): in function `Perl_get_ANYOFM_contents':
regcomp.c:(.text+0x8120): multiple definition of `Perl_get_ANYOFM_contents'; lib/auto/re/re.a(re_comp.o):re_comp.c:(.text+0xab00): first defined here
/usr/bin/ld: libperl.a(regcomp.o): in function `Perl_get_ANYOFHbbm_contents':
regcomp.c:(.text+0x81f0): multiple definition of `Perl_get_ANYOFHbbm_contents'; lib/auto/re/re.a(re_comp.o):re_comp.c:(.text+0xabd0): first defined here
/usr/bin/ld: libperl.a(regcomp_invlist.o): in function `Perl_populate_bitmap_from_invlist':
regcomp_invlist.c:(.text+0x300): multiple definition of `Perl_populate_bitmap_from_invlist'; lib/auto/re/re.a(re_comp_invlist.o):re_comp_invlist.c:(.text+0xa0): first defined here
/usr/bin/ld: libperl.a(regcomp_invlist.o): in function `Perl_populate_invlist_from_bitmap':
regcomp_invlist.c:(.text+0x13f0): multiple definition of `Perl_populate_invlist_from_bitmap'; lib/auto/re/re.a(re_comp_invlist.o):re_comp_invlist.c:(.text+0x300): first defined here
collect2: error: ld returned 1 exit status
make[3]: *** [makefile:393: perl] Error 1
make[3]: Leaving directory '/home/openwrt/musl/build_dir/hostpkg/perl/perl-5.38.0'
make[2]: *** [Makefile:146: /home/openwrt/musl/build_dir/hostpkg/perl/perl-5.38.0/.built] Error 2
make[2]: Leaving directory '/home/openwrt/musl/feeds/packages/lang/perl'
time: package/feeds/packages/perl/host-compile#284.63#28.19#310.96
    ERROR: package/feeds/packages/perl [host] failed to build.
make[1]: *** [package/Makefile:118: package/feeds/packages/perl/host/compile] Error 1
make[1]: Leaving directory '/home/openwrt/musl'
make: *** [/home/openwrt/musl/include/toplevel.mk:225: package/perl/compile] Error 2
```

--- a/ext/re/Makefile.PL
+++ b/ext/re/Makefile.PL
@@ -28,6 +28,13 @@ foreach my $tuple (@files) {
 }
 
 my $defines = '-DPERL_EXT_RE_BUILD -DPERL_EXT_RE_DEBUG -DPERL_EXT';
+my %args;
+for my $arg (@ARGV) {
+    $args{$1} = $2 if $arg =~ /^(\w+)=(.*)$/;
+}
+if ($args{LINKTYPE} eq "static") {
+    $defines .= ' -DPERL_EXT_RE_STATIC';
+}
 
 my @libs;
 if ($^O eq 'cygwin' && $Config{usequadmath}) {

--- a/regcomp.h
+++ b/regcomp.h
@@ -1559,12 +1559,12 @@ typedef enum {
  * and this is not a DEBUGGING enabled build (identified by
  * DEBUGGING_RE_ONLY being defined)
  */
-#if ( defined(USE_DYNAMIC_LOADING) && defined(DEBUGGING)) || \
+#if ( !defined(PERL_EXT_RE_STATIC) && defined(DEBUGGING)) ||       \
     ( defined(PERL_EXT_RE_BUILD) && defined(DEBUGGING_RE_ONLY)) || \
     (!defined(PERL_EXT_RE_BUILD) && defined(DEBUGGING))
 #define PERL_RE_BUILD_DEBUG
 #endif
-#if ( defined(USE_DYNAMIC_LOADING) || !defined(PERL_EXT_RE_BUILD) )
+#if !defined(PERL_EXT_RE_STATIC)
 #define PERL_RE_BUILD_AUX
 #endif
 

